### DISCUSSION TITLE
103. InnoDB에서 레코드 락을 인덱스에 거는 이유가 무엇일까요?

### DISCUSSION THREAD

**Discussion by jongmee** [jewelry-box/cs-interview#107] [General]
키워드: 레코드 락, 레코드 수준 잠금

**Comment by cutehumanS2** [comment]
InnoDB에서 인덱스에 레코드 락을 거는 이유는 **잠금 범위를 최소화하여 동시성 처리 성능을 높이고**, **데이터 정합성을 보장하기 위함**입니다.

- **잠금 범위를 최소화하여 동시성 처리 성능을 높입니다.**
    인덱스를 사용하면 쿼리 조건에 맞는 특정 레코드에만 잠금을 설정할 수 있어, 불필요한 레코드에 대한 잠금을 방지하고 동시성 처리 성능을 향상시킵니다.
    
- **데이터 정합성을 보장합니다.**
    인덱스를 기반으로 갭 락과 넥스트 키락을 설정하여 데이터의 중복 삽입을 방지하고, 트랜잭션 간 데이터 정합성을 유지할 수 있습니다.
    
<br>

[참고한 자료]
책 <Real MySQL 8.0 (1권)> - 5.3 InnoDB 스토리지 엔진 잠금
